File: backend/src/main/java/com/example/demo/models/Certificate.java
package com.example.demo.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "certificates")
@Data 
@NoArgsConstructor
@AllArgsConstructor
public class Certificate {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String name;
    private String description;
    private String issuedBy;
    private String city;
    private String place;
    private String eventName;
    private String eventType;
    private String fileUrl;
    private String imageUrl;
    private String issueDate;
    
    @Column(nullable = false)
    private Integer importance = 0;

    @Column(name = "is_pinned", nullable = false)
    private boolean isPinned = false;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;
}

File: frontend/src/types/certificate.ts
export interface Certificate {
  id: number;
  title?: string | null;
  name: string;
  issuedBy: string;
  description?: string | null;
  city?: string | null;
  place?: string | null;
  eventName?: string | null;
  eventType?: string | null;
  fileUrl?: string | null;
  imageUrl: string;
  issueDate?: string | null;
  importance?: number | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  pinned?: boolean;
  isPinned?: boolean;
}

export interface CreateCertificatePayload {
  name: string;
  issuedBy: string;
  imageUrl: string;
  issueDate?: string;
  description?: string;
  city?: string;
  place?: string;
  eventName?: string;
  eventType?: string;
  importance?: number;
  pinned?: boolean;
}
